server.R

```diff
@@ -50,17 +83,37 @@ server <- function(input, output,session) {
     if(input$subtype == "Multiplot" && length(input$genes) > 0) {
       query <- geneQuery(genes = input$genes, table = input$dataset)
       tcga <- dbGetQuery(database,query)
-
       query <- clinicalQuery(factors = "*", dataset = input$dataset)
       clinical <- dbGetQuery(database,query)
       clinical <- merge(tcga, clinical, by="UPN")
 
-      g <- ggplot(clinical,aes(fill=Gene, y=Expression, x=UPN, text = paste0("UPN ID: ",UPN,"<br />Dataset: ", input$dataset))) + geom_bar(position="dodge", stat="identity") + theme_bw() +
-        theme(text=element_text(size=12, family="avenir", face="bold"), axis.text=element_text(size=10, family="avenir", face="bold"),
-              axis.title=element_text(size=12, family="avenir", face="bold"),
-              axis.text.x = element_text(angle = 90, hjust = 1)) +
-        ggtitle("Multiple gene view") +
-        ylab("Log2 Expression") + xlab("")
+      # Render log2/raw graph based on user input
+      if (show.raw()){
```

Comment:
probably make the boolean "show.log2" to match the above changes and modify logic as needed